src/EnergyPlus/PlantUtilities.cc

```diff
@@ -2059,4 +2059,9 @@ MinFlowIfBranchHasVSPump(EnergyPlusData &state, PlantLocation const &plantLoc, b
     return branchPumpMinFlowLimit;
 }
 
+bool okToIssueWarning(EnergyPlusData &state, PlantLocation const &plantLoc)
+{
+    return (state.dataPlnt->PlantLoop(plantLoc.loopNum).LoopSide(plantLoc.loopSideNum).FlowLock != DataPlant::FlowLock::Unlocked &&
+            !state.dataGlobal->WarmupFlag);
+}
```

Comment:
I don't see it in CppCheck results but it sure looks like the state argument can be declared as const? Maybe you tried that?